### ISSUE TITLE
Doesn't converge when I train with my own data

### ISSUE THREAD

**Issue by zf-666** [TencentARC/BrushNet#67]
Loss has been shaking.， Wish I could see a picture of the correct loss，

**Comment by CharlesGong12** [comment]
Me too. It confuses me a lot. Have you solved it? @zf-666  Or could the authors help us please? @yxgeee @wondervictor @xinntao @bluestyle97

**Comment by SmileTAT** [comment]
How did you build your dataset?

**Comment by MrWH123** [comment]
hi @juxuan27 @yuanhangio, thanks for such great work! 
Could you plz share some details on training Brushnet_sdxl? such as how many epochs, how long it takes, how many GPU used?
I just use **one** zip package in BrashData just want to realize the training detail. Butt I find the loss fluctuation even if 11000+ steps passed (one zip has 10000 images, with batch size 4, around 4epoch). 
﻿
I guess the fluctuation loss is related to the random timestep during training. but how to  identify when the model is converged if loss has little guidance?
﻿
![image](https://github.com/user-attachments/assets/6f37c616-f2b4-4747-9bd3-3ff8dfd72772)
﻿
![image](https://github.com/user-attachments/assets/893b5350-5c34-480a-a896-b5e545e32e38)
﻿
similar issue seem in (https://github.com/TencentARC/BrushNet/issues/35), but didn't find some explanation on the loss

**Comment by zf-666** [comment]
> I fix the problem by use 'fp16' and 'fp16 vae' , but another problem arises, my dataset is on the dark side, but the resulting data, while fitting the distribution, is always on the light side

**Comment by xduzhangjiayu** [comment]
> hi @juxuan27 @yuanhangio, thanks for such great work! Could you plz share some details on training Brushnet_sdxl? such as how many epochs, how long it takes, how many GPU used? I just use **one** zip package in BrashData just want to realize the training detail. Butt I find the loss fluctuation even if 11000+ steps passed (one zip has 10000 images, with batch size 4, around 4epoch). ﻿ I guess the fluctuation loss is related to the random timestep during training. but how to identify when the model is converged if loss has little guidance? ﻿ ![image](https://private-user-images.githubusercontent.com/44314070/359372369-6f37c616-f2b4-4747-9bd3-3ff8dfd72772.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyMDgwODMsIm5iZiI6MTcyNDIwNzc4MywicGF0aCI6Ii80NDMxNDA3MC8zNTkzNzIzNjktNmYzN2M2MTYtZjJiNC00NzQ3LTliZDMtM2ZmOGRmZDcyNzcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIxVDAyMzYyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkyYmY2M2EwYmZiOThkOTQwMThkZTEwMTU5MzM0OWU5ZDRiODU0NTE0YmU1OTBiZDZhZmQyODEzNDhkZTczMDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qXxanDkrWsrc1OwGZABCAu4-nEKfKCtocuFILI7Hb8w) ﻿ ![image](https://private-user-images.githubusercontent.com/44314070/359371625-893b5350-5c34-480a-a896-b5e545e32e38.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyMDgwODMsIm5iZiI6MTcyNDIwNzc4MywicGF0aCI6Ii80NDMxNDA3MC8zNTkzNzE2MjUtODkzYjUzNTAtNWMzNC00ODBhLWE4OTYtYjVlNTQ1ZTMyZTM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIxVDAyMzYyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0MjFhZWY2MWIyZGQ3MjZiZGEyZDZmNDFjZmE4N2UwNDNmM2Q2MGVkM2MyM2RiMDBkZDUzYTA2ZGQ2MWQ2OTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.acc0644vDVpyiJY19oYCQ1XvZK8-7qCLG672MwIy_ro) ﻿ similar issue seem in (#35), but didn't find some explanation on the loss

Hi, which resolution of images did you used for training? Only 1024x1024 or random resolution? Appreciate for the reply!

**Comment by MrWH123** [comment]
> > hi @juxuan27 @yuanhangio, thanks for such great work! Could you plz share some details on training Brushnet_sdxl? such as how many epochs, how long it takes, how many GPU used? I just use **one** zip package in BrashData just want to realize the training detail. Butt I find the loss fluctuation even if 11000+ steps passed (one zip has 10000 images, with batch size 4, around 4epoch). ﻿ I guess the fluctuation loss is related to the random timestep during training. but how to identify when the model is converged if loss has little guidance? ﻿ ![image](https://private-user-images.githubusercontent.com/44314070/359372369-6f37c616-f2b4-4747-9bd3-3ff8dfd72772.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyMDgwODMsIm5iZiI6MTcyNDIwNzc4MywicGF0aCI6Ii80NDMxNDA3MC8zNTkzNzIzNjktNmYzN2M2MTYtZjJiNC00NzQ3LTliZDMtM2ZmOGRmZDcyNzcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIxVDAyMzYyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkyYmY2M2EwYmZiOThkOTQwMThkZTEwMTU5MzM0OWU5ZDRiODU0NTE0YmU1OTBiZDZhZmQyODEzNDhkZTczMDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qXxanDkrWsrc1OwGZABCAu4-nEKfKCtocuFILI7Hb8w) ﻿ ![image](https://private-user-images.githubusercontent.com/44314070/359371625-893b5350-5c34-480a-a896-b5e545e32e38.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyMDgwODMsIm5iZiI6MTcyNDIwNzc4MywicGF0aCI6Ii80NDMxNDA3MC8zNTkzNzE2MjUtODkzYjUzNTAtNWMzNC00ODBhLWE4OTYtYjVlNTQ1ZTMyZTM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIxVDAyMzYyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0MjFhZWY2MWIyZGQ3MjZiZGEyZDZmNDFjZmE4N2UwNDNmM2Q2MGVkM2MyM2RiMDBkZDUzYTA2ZGQ2MWQ2OTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.acc0644vDVpyiJY19oYCQ1XvZK8-7qCLG672MwIy_ro) ﻿ similar issue seem in (#35), but didn't find some explanation on the loss
> 
> Hi, which resolution of images did you used for training? Only 1024x1024 or random resolution? Appreciate for the reply!

1024x1024 for SDXL

**Comment by MrWH123** [comment]
> > I fix the problem by use 'fp16' and 'fp16 vae' , but another problem arises, my dataset is on the dark side, but the resulting data, while fitting the distribution, is always on the light side

could you share your training hyper-parameters and loss figure?

**Comment by shaoyandea** [comment]
@yuanhangio @juxuan27 there are only about 5-10 images in own data set, can brushnet converge?? how many images at leat should I prepare??

**Comment by xduzhangjiayu** [comment]
I use `BrushData` as my dataset, but in some data missing "width" in .tar file, so the training process is failed. Is anyone know how to fix this in `train_brushnet.py` to skip this sample and continue training? 
Thanks so much!

**Comment by xduzhangjiayu** [comment]
> > > I fix the problem by use 'fp16' and 'fp16 vae' , but another problem arises, my dataset is on the dark side, but the resulting data, while fitting the distribution, is always on the light side
> 
> could you share your training hyper-parameters and loss figure?

Hi, did you solve it? Could you please share some results images? Thanks!
